pandora-server-directory/src/lifecycle.ts

```diff
@@ -17,25 +17,35 @@ const logger = GetLogger('Lifecycle');
 	wtfnode.setLogger('error', (...message) => wtfNodeLogger.error(...message));
 }
 
+let destroying: string | undefined;
 let stopping: Promise<void> | undefined;
 const STOP_TIMEOUT = 10_000;
 
+function DestroyService(service: Service & object): Promise<void> | void {
```

Comment:
I don't think `& object` needs to be here
```suggestion
function DestroyService(service: Service): Promise<void> | void {
```